app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -260,8 +262,9 @@ class HomeFragment : Fragment() {
                     //  to some state.
                     recentTabs = getRecentTabs(components),
                     recentHistory = emptyList()
-                ),
-                listOf(
+                ).run { blocklistMiddleware.filterState(this) },
```

Comment:
This took some rather substantial refactoring, since `fitlerState` also relied on the handful of extension functions that were defined to filter the various state properties. I've left it as a separate commit for now, in case there's desire to see the diff between the two versions.

> We don't want other parts of the app to invoke it directly, see also my comment on the middleware as to why. 

Which comment is that? It makes sense to keep the responsibility of the middleware limited in that way to me in any case. This was mostly a side-effect of trying to keep everything initially encapsulated in the middleware class, and the original design clearly wasn't flexible enough for the end goal.